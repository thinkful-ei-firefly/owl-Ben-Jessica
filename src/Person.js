import React from 'react';

function Person(props) {
    let status = 'left session'
    if (props.inSession) {status = 'in session'}
    if (props.onStage) {status = 'on stage'}

    return (
      
        <div id={props.id}>
            <img src={props.avatar} alt='icon'/>
            <span className='personName'>{props.name}</span>
            <span className={props.inSession ? 'green' : 'gray'}>&#11044;</span>
            <span>{status}</span>
          
        </div>
     
    );
  }
  
  export default Person;