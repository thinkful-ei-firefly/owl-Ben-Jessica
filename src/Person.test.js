import React from 'react';
import ReactDOM from 'react-dom';
import store from './store'
import renderer from 'react-test-renderer';

import Person from './Person';

describe('<Stage />', () => {
    const part = store.participants;
    const person = part[0];
    it('renders without crashing', () => {
        // Create a DOM element to render the component into
        const div = document.createElement('div');

        // Render the component
        // If something is wrong it will fail here
        ReactDOM.render(<Person
            key={person.id}
            avatar={person.avatar}
            name={person.name}
            inSession={person.inSession}
            onStage={person.onStage}
          />, div);

        // Clean up
        ReactDOM.unmountComponentAtNode(div);
    });
    it('renders this UI as expected', () => {
        // Render the component, as JSON
        const tree = renderer.create(<Person
            key={person.id}
            avatar={person.avatar}
            name={person.name}
            inSession={person.inSession}
            onStage={person.onStage}
          />).toJSON();
        // Check whether it matches the previous snapshot
        // Stored in __snapshots__/App.test.js.snap
        expect(tree).toMatchSnapshot(); 
    });
});