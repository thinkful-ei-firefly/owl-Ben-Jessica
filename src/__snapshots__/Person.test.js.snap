// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Stage /> renders this UI as expected 1`] = `
<div>
  <img
    alt="icon"
    src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
  />
  <span
    className="personName"
  >
    Koren Templeton
  </span>
  <span
    className="green"
  >
    â¬¤
  </span>
  <span>
    on stage
  </span>
</div>
`;
